mlrun/run.py

```diff
@@ -1218,3 +1226,610 @@ def wait_for_runs_completion(runs: list, sleep=3, timeout=0, silent=False):
         runs = running
 
     return completed
+
+
+class ArtifactType(Enum):
+    """
+    Possible artifact types to log using the MLRun `function_decorator`.
+    """
+
+    DATASET = "dataset"
+    DIRECTORY = "directory"
+    FILE = "file"
+    OBJECT = "object"
+    PLOT = "plot"
+    RESULT = "result"
+
+
+class _ContextHandler:
+    """
+    Private class for handling an MLRun context of a function that is wrapped in MLRun's `function_decorator`.
```

Comment:
same

Reply:
regarding the "proper documentation in our docs", I will write it once 1.1 is out as the docs are updated regardless so I can't publish it with this commit.